### PR TITLE
CDAP-3749 db user does not require a password

### PR DESCRIPTION
**Pull request by albertshau**
Fix the database plugins so that specifying a user does not
require also specifying a password. It is completely valid,
though perhaps a bad idea, for a user to have no password.

### CODE REVIEW

File: database-plugins/src/test/java/co/cask/hydrator/plugin/db/batch/sink/BatchETLDBTestRun.java [hunk 1]
```diff
@@ -362,6 +346,17 @@ public void testUserNamePasswordCombinations() throws Exception {
...
+    Map<String, String> nullPassword = new HashMap<>(baseSourceProps);
+    emptyPassword.put(Properties.DB.USER, "emptyPwdUser");
+    database = new ETLStage("databaseSource", new Plugin("Database", nullPassword));
+    etlConfig = ETLBatchConfig.builder("* * * * *")
+      .setSource(database)
+      .addSink(table)
+      .addConnection(database.getName(), table.getName())
+      .build();
+    appRequest = new AppRequest<>(ETLBATCH_ARTIFACT, etlConfig);
+    TestBase.deployApplication(appId, appRequest);
```
Comment (bdmogal) [inline database-plugins/src/test/java/co/cask/hydrator/plugin/db/batch/sink/BatchETLDBTestRun.java]: for consistency, should we add a test using DBSink as well? Also, not sure why `deployApplication` in this class needs to be called with the class name

**Comment by bdmogal** [comment]
minor comment. Also, do we have a test for empty password? Other than that, LGTM

File: database-plugins/src/main/java/co/cask/hydrator/plugin/DBManager.java [hunk 2]
```diff
@@ -47,9 +47,6 @@ public void validateJDBCPluginPipeline(PipelineConfigurer pipelineConfigurer, St
     Preconditions.checkArgument(!(dbConfig.user == null && dbConfig.password != null),
                                 "user is null. Please provide both user name and password if database requires " +
                                   "authentication. If not, please remove password and retry.");
```
Comment (gsps1) [inline database-plugins/src/main/java/co/cask/hydrator/plugin/DBManager.java:49]: looks like the username is always required, why can't we make that field non-nullable ?

**Comment by albertshau** [comment]
yep empty password is tested right above that section

File: database-plugins/src/main/java/co/cask/hydrator/plugin/DBManager.java (hunk 2 above)
Comment (albertshau) [inline database-plugins/src/main/java/co/cask/hydrator/plugin/DBManager.java:49]: its not always required, you can connect to databases without a username

Comment (gsps1) [inline database-plugins/src/main/java/co/cask/hydrator/plugin/DBManager.java:49]: oh, this check checks for password but no username, alright.

Comment (albertshau) [inline database-plugins/src/main/java/co/cask/hydrator/plugin/DBManager.java:49]: yep

**Comment by bdmogal** [comment]
:+1:
